### PR TITLE
Controls rework

### PR DESCRIPTION
**Pull request by tombatossals**
Reworked the controls functionality:

* Easier to extend, similar to layers (isPluginLoaded, checkValidParams, createControl "interface" methods).
* controls is now watched (watchCollection), so new controls can be added/removed dinamically, or controls now can wait until the map is initialized before being inserted.
* New and updated examples.

### CODE REVIEW

**Comment by elesdoar** [comment]
Always must be loaded the controls? so:

```javascript
leafletData.getMap().then(function(map) {
  angular.extend($scope.controls, {
    ...
  }); 
  ...
}
```

Or this can be loaded on the scope start?

**Comment by tombatossals** [comment]
Hi @elesdoar, in that example you reference we must wait until the map is initialized before adding the controls, that's for example the case of the minimap control, or the search-control.

In other cases, you doesn't need to wait for anything, of course.

**Comment by nmccready** [comment]
@tombatossals what are the breaking changes if any? Should this involve a bump to 0.9.X?

File: src/directives/controls.js [hunk 1]
```diff
@@ -10,81 +10,49 @@ angular.module("leaflet-directive").directive('controls', function ($log, leafle
...
-                createLayer = leafletLayerHelpers.createLayer,
-                leafletScope  = controller.getLeafletScope(),
-                controls = leafletScope.controls,
-                errorHeader = leafletHelpers.errorHeader + ' [Controls] ';
+            var createControl = leafletControlHelpers.createControl;
+            var isValidControlType = leafletControlHelpers.isValidControlType;
+            var leafletScope  = controller.getLeafletScope();
+            var isDefined = leafletHelpers.isDefined;
+            var leafletControls = {};
+            var errorHeader = leafletHelpers.errorHeader + ' [Controls] ';
```
Comment (nmccready) [inline src/directives/controls.js]: :+1:

File: examples/0404-controls-minimap-example.html [hunk 2]
```diff
@@ -27,20 +29,23 @@
...
-                        },
-                        toggleDisplay: true
-                    }
-                }
+                controls: {}
+           });
+
+           // Wait for center to be stablished
+           leafletData.getMap().then(function() {
+               angular.extend($scope.controls, {
```
Comment (nmccready) [inline examples/0404-controls-minimap-example.html]: nice

**Comment by nmccready** [comment]
LGTM, just waiting on feedback on the versioning

**Comment by tombatossals** [comment]
@nmccready no breaking changes theorically, now is preferred to pass the type of control as a property instead of as the key of the controls object, but setting the control type the old way will continue working.

If you want we can release these changes with the 0.9.x version, seems reasonable.

**Comment by nmccready** [comment]
OK 0.9 it is merging

**Pull request merged**